### PR TITLE
move funcs for getting gh tokens and clients

### PR DESCRIPTION
**Pull request by krancour**
There's a lot of logic in the `webhook` package for obtaining tokens for github and github clients that use those tokens. There's nothing webhook-specific about this code, so it seems a good candidate to be extracted into its own package... indeed, the check-run binary currently obtains its client from the `webhook` package and that doesn't feel right. This little bit of refactoring fixes that.

~~Note this will have merge conflicts with #81. If #81 (which removes a lot of dead code) is merged first, this PR will get a lot smaller when the merge conflicts are resolved.~~

Also note that `Gopkg.lock` changes are only because I used a new sub-package from testify in the new package's tests and that sub-package was previously unused by this project.

### CODE REVIEW

File: pkg/github/client_test.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	baseURL   = "http://example.com/base/"
+	uploadURL = "http://example.com/upload/"
+	testToken = "totallyFake"
+)
+
+func TestNewClientFromBearerToken(t *testing.T) {
+	ghc, err := NewClientFromBearerToken(baseURL, uploadURL, testToken)
+	require.NoError(t, err)
+	require.Equal(t, baseURL, ghc.BaseURL.String())
+	require.Equal(t, baseURL, ghc.BaseURL.String())
```
Comment (vdice) [inline pkg/github/client_test.go]: Duplicated assertion here -- should we be checking something else?  (`UploadURL`?)

File: pkg/github/client_test.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+func TestNewClientFromInstallationToken(t *testing.T) {
+	ghc, err := NewClientFromInstallationToken(baseURL, uploadURL, testToken)
+	require.NoError(t, err)
+	require.Equal(t, baseURL, ghc.BaseURL.String())
+	require.Equal(t, baseURL, ghc.BaseURL.String())
```
Comment (vdice) [inline pkg/github/client_test.go]: Duplicated assertion here -- should we be checking something else?  (`UploadURL`?)

File: pkg/github/client_test.go (hunk 1 above)
Comment (krancour) [reply pkg/github/client_test.go]: Yup! Nice catch. Will fix.

Comment (krancour) [reply pkg/github/client_test.go]: Fixed in https://github.com/brigadecore/brigade-github-app/pull/82/commits/1512905043e1e7384da2dc4c934ed0d4c762bf37

File: pkg/github/client_test.go (hunk 2 above)
Comment (krancour) [reply pkg/github/client_test.go]: Fixed in https://github.com/brigadecore/brigade-github-app/pull/82/commits/1512905043e1e7384da2dc4c934ed0d4c762bf37

**Pull request merged**